File: Ch 1.Arrays And Strings/9.String Rotation/StringRotation.java
import java.util.*;
public class StringRotation{
   public static void main(String[] args) {

    StringRotation array = new StringRotation();
    array.isRotation("waterbottle","erbottlewat");
  
    }

    public boolean isRotation(String s1, String s2)
    {
        if (s1.length() == s2.length())
            return (s1+s1).contains(s2);
        return false;


    }
}

File: Ch 1.Arrays And Strings/9.String Rotation/String Rotation.cpp


bool isRotation( std::string s1, std::string s2 ) {
    if (s1.size() != s2.size())
        return true;
    s1 += s2;    
    
    for(int i = 0;i < s2.size();i++){
        if(s1.substr(i,s2.size()) == s2){
                return true;           
            }
    }
    return false;
}
